pytorch_lightning/accelerators/cpu.py

```diff
@@ -48,3 +57,18 @@ def auto_device_count() -> int:
     def is_available() -> bool:
         """CPU is always available for execution."""
         return True
+
+
+def get_cpu_process_metrics() -> dict[str, float]:
+    if not _PSUTIL_AVAILABLE:
+        raise ModuleNotFoundError(
```

Comment:
This is a breaking change for existing device stats users who upgrade lightning. 
I propose alternative options:
1) Make it a warning, skip logging device stats unless they have the package installed
2) Turn it off by default
3) Add psutil to base requirements (not preferred)

In any case, the PR would need a changelog

Reply:
I'd prefer to go with 4) if that's good with you. If someone wants CPU metrics and uses `DeviceStatsMonitor` while running with only CPU, it would be a pain if no logging occurred and the user needed to look through the logs to figure out why. Additionally, if someone calls `get_cpu_process_metrics` they likely wouldn't expect to receive an empty dictionary.

Reply:
personally, I'd like to have 2) if possible.

maybe we can turn-if off by default, by either:
1. Changing `cpu_stats` to `add_cpu_stats`, which by default is False, but adds CPU stats for CPU accelerator and not for others.
2. default `cpu_stats=None`. If None or True and device is CPU then CPU stats are logged, else with other accelerators CPU stats are not logged. IF False, no CPU stats are logged in either case.

Reply:
I'll go with 2

Reply:
Actually, I don't see how this solves the issue of this being a breaking change. If `cpu_stats=None` and the CPU is used and `psutil` isn't installed, then this is a breaking change still

Reply:
yes, only if they add a callback, else it won't break other code-bases. not a breaking change but a feature since CPU accelerator never supported logging device stats. It would be a breaking change for other accelerators where `DeviceStatsMonitor` was working fine up until now but would break due to this.

Reply:
This would still be a breaking change if the user previously used the `DeviceStatsMonitor` callback and had a training script they used both with/without a GPU (or another accelerator). Previously, if they ran their code on CPU only, the `DeviceStatsMonitor` would just do nothing. However, now if they run their code on CPU only (leaving in `DeviceStatsMonitor`) and don't have `psutil` installed, they will get an exception

Reply:
okay. it should have raised some error before in that case, but we can't do anything for that now. but at least it should be optional now for other accelerators IMO and should not fail there.

Reply:
@awaelchli @rohitgr7 I am leaning towards option 2) here.

> Turn it off by default

If the user intends to have `Trainer(callbacks=DeviceStatsMonitor())` for the CPU Accelerator, we should raise an exception and request to install psutil.

Reply:
> what about logging cpu_stats with other accelerators?

`cpu_stats` will be False by default. If it's required, user could do `DeviceStatsMonitor(cpu_stats=True)`.

Reply:
> cpu_stats: if ``True``, will also record CPU stats in addition to accelerator stats.

"extra emphasis on addition to accelerator stats"

It would be bizarre behavior if user is doing `Trainer(DeviceStatsMonitor(cpu_stats=False))` because it would do absolutely nothing. 

If the `DeviceStatsMonitor` callback is passed to Trainer, the accelerator stats should be logged.

Reply:
yeah, that's why I want to confirm, should it be false by default as you mentioned?
```py
cpu_stats will be False by default
```

Reply:
yup, will push the changes once we come to the decision